Source/WebKit/WebProcess/WebPage/wpe/WebPageWPE.cpp

```diff
@@ -48,38 +48,34 @@ bool WebPage::platformCanHandleRequest(const ResourceRequest&)
 
 bool WebPage::hoverSupportedByPrimaryPointingDevice() const
 {
-#if ENABLE(TOUCH_EVENTS)
-    return !screenIsTouchPrimaryInputDevice();
-#else
-    return true;
-#endif
+    // For now, there's no concept of primary vs any
+    return WebProcess::singleton().availableInputDevices().contains(AvailableInputDevices::Mouse);
 }
 
 bool WebPage::hoverSupportedByAnyAvailablePointingDevice() const
 {
-#if ENABLE(TOUCH_EVENTS)
-    return !screenHasTouchDevice();
-#else
-    return true;
-#endif
+    return WebProcess::singleton().availableInputDevices().contains(AvailableInputDevices::Mouse);
 }
 
 std::optional<PointerCharacteristics> WebPage::pointerCharacteristicsOfPrimaryPointingDevice() const
 {
-#if ENABLE(TOUCH_EVENTS)
-    if (screenIsTouchPrimaryInputDevice())
+    const auto& availableInputs = WebProcess::singleton().availableInputDevices();
+    if (availableInputs.contains(AvailableInputDevices::Mouse))
+        return PointerCharacteristics::Fine;
+    if (availableInputs.contains(AvailableInputDevices::TouchScreen))
```

Comment:
Which is wat you are doing here, but this could check WebProcess::primaryPointingDevice() instead and the logic to device the primary is in one place.